Remade task to fix problems caused by rebase

adp-atea opened the pull request:
Remade task in new branch, fixed left over code from url remake in detail template

adp-atea committed:
Remade task in new branch, fixed left over code from url remake in detail template

adp-atea committed:
Removed non-existent migration in dependencies

adp-atea committed:
Update views.py

adp-atea committed:
Merge branch 'devel' into 43-duomenų-rinkinio-narių-sąrašas-1

# Conflicts:
#	vitrina/datasets/views.py

adp-atea committed:
Merged and fixed slugify todo

adp-atea committed:
Merge remote-tracking branch 'origin/43-duomenų-rinkinio-narių-sąrašas-1' into 43-duomenų-rinkinio-narių-sąrašas-1

sirex commented on vitrina/datasets/models.py:
```diff
@@ -128,6 +128,27 @@ def get_absolute_url(self):
+        return reverse('dataset-members', kwargs={'pk': self.pk})
+
+
+class DatasetMember(models.Model):
```
Please see #171 task, dataset representatives should be implemented using generic `Representative` model, please consult and collaborate with the #171 task author.

sirex commented on vitrina/datasets/templates/vitrina/datasets/detail.html:
```diff
@@ -22,11 +22,6 @@
 
 {% block content %}
```
Add dataset contextual navigation menu (tabs), similar to the one added to organization details template.

sirex commented on vitrina/datasets/services.py:
```diff
@@ -105,3 +106,7 @@ def can_create_dataset(user: User, org_id) -> bool:
+
+
+def can_see_dataset_members(user: User, dataset: Dataset) -> bool:
+    return Representative.objects.filter(organization_id=dataset.organization_id, email=user.email).exists()
```
`user.is_staff` and `user.is_superuser` should have access to this. Please collaborate with the #171.

sirex reviewed (changes requested):
There are no tests. All functionality must be covered with tests.

adp-atea committed:
Merge branch 'devel' into 43-duomenų-rinkinio-narių-sąrašas-1

# Conflicts:
#	vitrina/datasets/services.py
#	vitrina/datasets/views.py

adp-atea committed:
Remade with Representative

adp-atea committed:
Implemented tabs in dataset detail view

sirex reviewed (changes requested):
This branch is out of sync with devel branch and has conflicting changes with devel. Please update this branche with newest changes from devel and fix conflicts.

adp-atea committed:
Merge branch 'devel' into 43-duomenų-rinkinio-narių-sąrašas-1

# Conflicts:
#	tests/datasets/test_views.py
#	vitrina/datasets/services.py
#	vitrina/datasets/templates/vitrina/datasets/detail.html
#	vitrina/datasets/urls.py
#	vitrina/datasets/views.py

adp-atea committed:
Merged, fixed tabs not showing properly

sirex committed:
Remove unused template and fix a tab title

sirex committed:
Fix tabs and DatasetMembersView

Dataset, Request and Project views each has differet set of tabs, so it
is not a good idea to reuse one tabs.html template for all of them.
Currently I separated tabs.html only for Dataset.

Also refactored DatasetMebersView, wich acts like a list view, bet
extends DatasetDetailsView. Changed it to ListView as it supposed to be.

sirex committed:
Refactor OrganizationMembersView to ListView

Previously, OrganizationMembersView was based on OrganizationDetailView,
even if it is a list.

sirex committed:
Add CRUD for datasets members

sirex committed:
Fix failing tests

sirex committed:
Add tests for dataset members crud

sirex committed:
An attempt to fix flaky test

I didn't found a reason why tests are flaky, but one of ideas, that
categories ar sorted randomly,so maybe this had some effect on tests.
Added fixed titles.
